### PR TITLE
[SELC-3573] feat: Added function to reject outdated onboardings

### PR DESCRIPTION
**Pull request by pierpaolodidato89**
#### List of Changes

Added function to reject outdated onboardings

#### Motivation and Context

After onboarding's completion, a new function is invoked in order to reject (status = REJECTED) outdated onboardings.

#### How Has This Been Tested?

I have deployed the feature branch in DEV and tested the scenario creating two different onboardings for the same institution (without completing one of them)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by manuraf — COMMENTED** [review]
you could move this action away from an WorkFlow execution and using a scheduled Azure Functions such PurgeInstancesCompleted that runs once a day or a week...In addition, pay attention that this type of query requires a lot of Request Unit without adding a specific index.

**Comment by pierpaolodidato89** [comment]
> you could move this action away from an WorkFlow execution and using a scheduled Azure Functions such PurgeInstancesCompleted that runs once a day or a week...In addition, pay attention that this type of query requires a lot of Request Unit without adding a specific index.

I'm adding an index and working to increase performance of the query. Considering that the new function is part of async process,  I'm thinking to leave it into worflow execution

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java [hunk 1]
```diff
@@ -217,6 +218,15 @@ public void persistActivatedAt(Onboarding onboarding) {
                 .where("_id", onboarding.getId());
     }
 
+    @Override
+    public void rejectOutdatedOnboardings(Onboarding onboarding) {
+        LocalDateTime now = LocalDateTime.now();
+        onboardingRepository
+                .update("status = ?1 and updatedAt = ?2 ", REJECTED, now)
+                .where("productId = ?1 and institution.origin = ?2 and institution.originId = ?3 and _id != ?4 (status = PENDING or status = TOBEVALIDATED)",
```
Comment (andrea-putzu) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java:226]: Are we sure that an onaboarding pending have an institution.id?

Comment (pierpaolodidato89) [reply apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java:226]: Yes, coz this function is invoked after the CREATE INSTITUTION activity. Therefore, it's not possible to use the only condition on status=PENDING coz the update to COMPLETED happens only after the entire block `executePendingState`

**Comment by andrea-putzu** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=pagopa_selfcare-onboarding&pullRequest=502) **Quality Gate failed**
> Failed conditions  [D Maintainability Rating on New Code](https://sonarcloud.io/dashboard?id=pagopa_selfcare-onboarding&pullRequest=502) (required ≥ A)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=pagopa_selfcare-onboarding&pullRequest=502)
> 
>  Catch issues before they fail your Quality Gate with our IDE extension  [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=pull-request)

Sonar failed condition include analysis for overall code

**Comment by andrea-putzu** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=pagopa_selfcare-onboarding&pullRequest=502) **Quality Gate failed**
> Failed conditions  [B Maintainability Rating on New Code](https://sonarcloud.io/dashboard?id=pagopa_selfcare-onboarding&pullRequest=502) (required ≥ A)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=pagopa_selfcare-onboarding&pullRequest=502)
> 
>  Catch issues before they fail your Quality Gate with our IDE extension  [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=pull-request)

Sonar failed condition include analysis for overall code

**Pull request merged**